### PR TITLE
gradle extracts gatk-launch from the gatk jar

### PR DESCRIPTION
**Pull request by lbergelson**
adding a setupGatkLaunch task to gradle which extracts the gatk-launch script from the gatk dependency and places it in the root directory
updating gatk version

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -173,6 +173,20 @@ configurations {
...
 
+def findJarByName(Configuration config, String name) {
+    config.find { File file -> file.name.startsWith(name)}
+}
+
+task setupGatkLaunch(type: Copy) {
+    def gatkJar = { findJarByName(configurations.compile, 'gatk') }
+    from {
+        zipTree( gatkJar ).matching{ include 'gatk-launch'}
+    }
```
Comment (droazen) [inline build.gradle]: Will this silently continue if there's no `gatk-launch` inside the jar?

File: .gitignore [hunk 2]
```diff
@@ -39,3 +39,5 @@ foo.patch
 # odt temp file
 /docs/PoN/.~lock.TangentNormalizationInSparkNotes.odt#
 
+# gatk-launch
```
Comment (droazen) [inline .gitignore]: Remove uninformative comment? :)

File: build.gradle [hunk 3]
```diff
@@ -173,6 +173,20 @@ configurations {
...
+    into "$projectDir"
+    outputs.file "$projectDir/gatk-launch"
```
Comment (droazen) [inline build.gradle]: Should the String `gatk-launch` be a named constant in case we rename it?

File: build.gradle [hunk 4]
```diff
@@ -173,6 +173,20 @@ configurations {
     }
 }
 
+def findJarByName(Configuration config, String name) {
+    config.find { File file -> file.name.startsWith(name)}
+}
+
+task setupGatkLaunch(type: Copy) {
+    def gatkJar = { findJarByName(configurations.compile, 'gatk') }
```
Comment (droazen) [inline build.gradle]: Can you confirm that this won't match the `gatk-protected` jar itself?

**Comment by droazen** [comment]
Review complete, back to @lbergelson, merge after addressing comments

File: build.gradle (hunk 4 above)
Comment (lbergelson) [inline build.gradle]: I'm not sure how to check that actually...  I think it won't be a problem if it does because it won't have a gatk-launch.

Comment (lbergelson) [inline build.gradle]: well, we could check it by printlning I guess...

File: build.gradle (hunk 1 above)
Comment (lbergelson) [inline build.gradle]: fixed

File: .gitignore (hunk 2 above)
Comment (lbergelson) [inline .gitignore]: fixed

File: build.gradle (hunk 4 above)
Comment (lbergelson) [inline build.gradle]: verified, only gatk public jar

File: build.gradle (hunk 3 above)
Comment (lbergelson) [inline build.gradle]: fixed

**Pull request merged**